### PR TITLE
refactor(framework): adopt to eva mappings [partial]

### LINKED ISSUE #311: Adopt existing components to support text styles
+ apply new flat styles

### PR DESCRIPTION
**Pull request by 32penkin**
### Please read and mark the following check list before creating a pull request:

 - [x] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/react-native-ui-kitten/blob/master/CONTRIBUTING.md) guide.
 - [x] I read and followed the [New Feature Checklist](https://github.com/akveo/react-native-ui-kitten/blob/master/DEV_DOCS.md) guide.

 #### Short description of what this resolves:

Updates following components to eva mappigns:
* `Layout`
* `Modal`
* `OverflowMenu`
* `BottomTabNavigator`
* `TopNavigationBar`
* `Toggle`

This closes #311

### CODE REVIEW

File: src/framework/ui/modal/modal.component.tsx [hunk 1]
```diff
@@ -27,7 +24,7 @@ interface ModalProps {
 
 const { width, height } = Dimensions.get('window');
 
-export type Props = StyledComponentProps & ViewProps & ModalProps;
+export type Props = ViewProps & ModalProps;
```
Comment (artyorsh) [inline src/framework/ui/modal/modal.component.tsx]: Is it not styled anymore?

File: src/framework/ui/toggle/toggle.component.tsx [hunk 2]
```diff
@@ -133,44 +133,46 @@ export class Toggle extends React.Component<Props> {
...
     return {
       wrapper: {
-        width: container.width,
-        height: container.height,
+        width: style.width,
+        height: style.height,
       },
       componentContainer: {
-        ...container,
+        ...style,
```
Comment (artyorsh) [inline src/framework/ui/toggle/toggle.component.tsx]: really? all styles to one container, but some of them to `wrapper`?

File: src/framework/ui/modal/modal.component.tsx (hunk 1 above)
Comment (32penkin) [reply src/framework/ui/modal/modal.component.tsx]: You're right, modal is not a part of our system. It's just a component for using custom content.

File: src/framework/ui/toggle/toggle.component.tsx (hunk 2 above)
Comment (32penkin) [reply src/framework/ui/toggle/toggle.component.tsx]: Refactored, it was just an unused code.

**Comment by artyorsh** [comment]
Merged in #313

**Pull request closed without merging**